src/containers/Application/ApplicationSummary.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react'
+import { FullStructure, ResponsesByCode, ElementStateNEW } from '../../utils/types'
+import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { Loading, NoMatch } from '../../components'
+import { SectionWrapper } from '../../components/Application'
+import strings from '../../utils/constants'
+import { Button, Header, Message, Container } from 'semantic-ui-react'
+
+interface ApplicationProps {
+  structure: FullStructure
+}
+
+const ApplicationSummary: React.FC<ApplicationProps> = ({ structure }) => {
+  const { push } = useRouter()
+  const { error, fullStructure } = useGetFullApplicationStructure({
+    structure,
+  })
+
+  useEffect(() => {
+    if (!fullStructure) return
+    // Re-direct if application is not valid
+    if (fullStructure.info.firstStrictInvalidPage) {
+      const { sectionCode, pageNumber } = fullStructure.info.firstStrictInvalidPage
+      push(`/applicationNEW/${fullStructure.info.serial}/${sectionCode}/Page${pageNumber}`)
```

Comment:
i think all of the redirect should be replace

Reply:
Yeah, makes sense for re-directs.